work_queue/src/work_queue_json.c

```diff
@@ -0,0 +1,341 @@
+
+    while (key != NULL) {
+
+        if(!is_in(key, array)){
+            return 1;
+        }
+
+        key = jx_iterate_keys(json, &j);
+
+    }
+
+    return 0;
+
+}
+
+static int specify_files(int input, struct jx *files, struct work_queue_task *task){
+
+    void *i = NULL;
+    struct jx *arr = jx_iterate_array(files, &i);
+
+    while (arr != NULL){
+        
+        char *local, *remote;
+        int flags;
+        void *k = NULL;
+        void *v = NULL;
+
+        const char *key = jx_iterate_keys(arr, &k);
+        struct jx *value = jx_iterate_values(arr, &v);
+
+        while (key != NULL){
+
+
+            if (!strcmp(key, "local")){
+                local = value->u.string_value;
+            }
+            else if (!strcmp(key, "remote")){
+                remote = value->u.string_value;
+            }
+            else if (!strcmp(key, "flags")){
+                flags = value->u.integer_value;
+            }
+            else{
+                printf("%s\n",value->u.string_value);
+            }
+
+            key = jx_iterate_keys(arr, &k);
+            value = jx_iterate_values(arr, &v);
+
+        }
+
+        if (input){
+            work_queue_task_specify_file(task, local, remote, WORK_QUEUE_INPUT, flags);
+        }
+        else{
+            work_queue_task_specify_file(task, local, remote, WORK_QUEUE_OUTPUT, flags);
+        }
+
+        arr = jx_iterate_array(files, &i);
+    
+    }
+
+    return 0;
+
+}
+
+
+static struct work_queue_task* create_task(const char* str){
+
+    char *command_line;
+    struct jx *input_files, *output_files;
+
+    struct jx *json = jx_parse_string(str); 
+    if(!json){
+        return NULL;
+    }
+
+    //validate json
+    if(validate_json(json, work_queue_task_properties)){
+        return NULL;
+    }
+
+    //get command from json
+    void *j = NULL;
+    void *i = NULL;
+    const char* key = jx_iterate_keys(json, &j);
+    struct jx *value = jx_iterate_values(json, &i);
+
+    while (key != NULL) {
+
+        if(!strcmp(key, "command_line")){
+            command_line = value->u.string_value;
+        }
+        else if (!strcmp(key, "input_files")){
+            input_files = value;
+        }
+        else if (!strcmp(key, "output_files")){
+            output_files = value;
+        }
+        else{
+            printf("%s\n",value->u.string_value);
+        }
+
+        key = jx_iterate_keys(json, &j);
+        value = jx_iterate_values(json, &i);
+    
+    }    
+
+    //call work_queue_task_create
+    if(command_line){
+
+        struct work_queue_task* task = work_queue_task_create(command_line);
+
+        if (!task){
+            return NULL;
+        }
+
+        if(input_files){            
+            specify_files(1, input_files, task);
+        }
+
+        if(output_files){
+            specify_files(0, output_files, task);
+        }
+    
+        return task;
+
+    }
+
+    return NULL;
+
+}
+
+/** Create a new work_queue object.
```

Comment:
This is the right idea, except that the doc string goes in the header file, not the source file.
You can test the doc generation by doing a "make docs" at the top level, and then opening
doc/api/html/index.html in your web browser.